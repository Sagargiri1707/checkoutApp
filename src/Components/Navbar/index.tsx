function Navbar() {
  return (
    <header className="min-w-[1000px]">
      <div className="flex bg-TK-background text-white h-[60px]">
        <div className="flex items-center m-4">
          <div className="flex pr-3 pl-3">
            <div className="mt-7 text-xs xl:text-sm font-bold">
              Connect with me on
            </div>
          </div>
          <a target="_blank" href={'https://www.linkedin.com/in/sagargiri07'}>
            <div className="flex pr-3 pl-3">
              <div className="mt-7 text-xs xl:text-sm font-bold">Linkedin</div>
            </div>
          </a>
          <a target="_blank" href={'https://www.twitter.com/sagargiri1707'}>
            <div className="flex pr-3 pl-3">
              <div className="mt-7 text-xs xl:text-sm font-bold">Twitter</div>
            </div>
          </a>
        </div>
        <div className="flex pr-3 pl-3 absolute right-0">
          <div className="mt-7 text-xs xl:text-sm font-bold">Sagar Giri</div>
        </div>
      </div>
    </header>
  );
}

export default Navbar;
