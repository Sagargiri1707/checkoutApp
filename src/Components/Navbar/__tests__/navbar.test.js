import '@testing-library/jest-dom';
import React from 'react';
import { render, screen } from '@testing-library/react';
import Navbar from '../index';
import { Link } from 'react-router-dom';
jest.mock('react-router-dom', () => {
  const ActualReact = jest.requireActual('react-router-dom');
  return {
    ...ActualReact,
    Link: jest.fn(),
  };
});
describe('navbar component', () => {
  it('should render a header ', () => {
    const container = render(<Navbar />);
    expect(container).toMatchSnapshot();
  });
});
