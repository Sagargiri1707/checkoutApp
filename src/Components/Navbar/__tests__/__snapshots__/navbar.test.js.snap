// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navbar component should render a header  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header>
        <div
          class="flex bg-TK-background text-white h-[60px] items-center justify-between	"
        >
          <div
            class="flex items-center m-4"
          />
          <div
            class="flex pr-3 pl-3 hidden lg:flex"
          >
            <a
              href="https://www.linkedin.com/in/sagargiri07"
              target="_blank"
            >
              <div
                class="flex pr-3 pl-3"
              >
                <div
                  class="mt-7 text-xs xl:text-sm font-bold"
                >
                  Linkedin
                </div>
              </div>
            </a>
            <a
              href="https://www.twitter.com/sagargiri1707"
              target="_blank"
            >
              <div
                class="flex pr-3 pl-3"
              >
                <div
                  class="mt-7 text-xs xl:text-sm font-bold"
                >
                  Twitter
                </div>
              </div>
            </a>
          </div>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header>
      <div
        class="flex bg-TK-background text-white h-[60px] items-center justify-between	"
      >
        <div
          class="flex items-center m-4"
        />
        <div
          class="flex pr-3 pl-3 hidden lg:flex"
        >
          <a
            href="https://www.linkedin.com/in/sagargiri07"
            target="_blank"
          >
            <div
              class="flex pr-3 pl-3"
            >
              <div
                class="mt-7 text-xs xl:text-sm font-bold"
              >
                Linkedin
              </div>
            </div>
          </a>
          <a
            href="https://www.twitter.com/sagargiri1707"
            target="_blank"
          >
            <div
              class="flex pr-3 pl-3"
            >
              <div
                class="mt-7 text-xs xl:text-sm font-bold"
              >
                Twitter
              </div>
            </div>
          </a>
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
